### PR TITLE
document cflags argument to BPF()

### PR DESCRIPTION
**Pull request by ahrens**
This relates to https://github.com/iovisor/bcc/issues/1707

### CODE REVIEW

File: docs/reference_guide.md [hunk 1]
```diff
@@ -758,7 +758,7 @@ Constructors.
 
 ### 1. BPF
 
-Syntax: ```BPF({text=BPF_program | src_file=filename} [, usdt_contexts=[USDT_object, ...]])```
+Syntax: ```BPF({text=BPF_program | src_file=filename} [, usdt_contexts=[USDT_object, ...]] [, cflags=[arg1, ...]])```
```
Comment (yonghong-song) [inline docs/reference_guide.md]: since you change it. How about make it complete, include `debug` and `hdr_file` as well?

Comment (ahrens) [reply docs/reference_guide.md]: Happy to, if you can explain to me what they are.

**Comment by brendangregg** [comment]
I'm not sure, so I checked from src/python/bcc/__init__.py:

<pre>
    def __init__(self, src_file=b"", hdr_file=b"", text=None, debug=0,
            cflags=[], usdt_contexts=[]):
        """Create a new BPF module with the given source code.

        Note:
            All fields are marked as optional, but either `src_file` or `text`
            must be supplied, and not both.

        Args:
            src_file (Optional[str]): Path to a source file for the module
            <b>hdr_file</b> (Optional[str]): Path to a helper header file for the `src_file`
            text (Optional[str]): Contents of a source file for the module
            <b>debug</b> (Optional[int]): Flags used for debug prints, can be |'d together
                                   See "Debug flags" for explanation
[...]
# Debug flags

# Debug output compiled LLVM IR.
DEBUG_LLVM_IR = 0x1
# Debug output loaded BPF bytecode and register state on branches.
DEBUG_BPF = 0x2
# Debug output pre-processor result.
DEBUG_PREPROCESSOR = 0x4
# Debug output ASM instructions embedded with source.
DEBUG_SOURCE = 0x8
#Debug output register state on all instructions in addition to DEBUG_BPF.
DEBUG_BPF_REGISTER_STATE = 0x10
</pre>

I don't know the purpose of hdr_file.

**Comment by drzaeus77** [comment]
```
            src_file (Optional[str]): Path to a source file for the module
            hdr_file (Optional[str]): Path to a helper header file for the `src_file`
```
These are both holdovers from src/cc/frontends/b, and shouldn't be used when using restricted-c style programs.

**Comment by ahrens** [comment]
@yonghong-song I added documentation of the debug flags.

**Comment by yonghong-song** [comment]
[buildbot, ok to test]

**Comment by yonghong-song** [comment]
Looks good. Thanks for help improving our documentation!

**Pull request merged**